ERC721/serverless-NFT/1-06.md

```diff
@@ -0,0 +1,211 @@
+        var totalGas = new BigNumber(0)
+
+        // Deploy Token.sol
+        token = await Token.new()
+        var tx = await web3.eth.getTransactionReceipt(token.transactionHash)
+        totalGas = totalGas.plus(tx.gasUsed)
+        console.log(_ + tx.gasUsed + ' - Deploy Token')
+        token = await Token.deployed()
+
+        console.log(_ + '-----------------------')
+        console.log(_ + totalGas.toFormat(0) + ' - Total Gas')
+        done()
+      } catch (error) {
+        console.error(error)
+        done(false)
+      }
+    })()
+  })
+
+  describe('Token.sol', function() {
+    it('should pass', async function() {
+      assert(
+        true === true,
+        'this is true'
+      )
+    })
+
+  })
+})
+```
+
+Import the Metadata at the top of the file then duplicate the token test code and replace it with Metadata so Metadata is imported first.
+
+<!-- TODO: What does this mean?  -->
+
+Also don't forget to set the deploy parameters for the token, including the metadata address.
+
+```javascript
+var Metadata = artifacts.require('./Metadata.sol')
+var Token = artifacts.require('./Token.sol')
+var BigNumber = require('bignumber.js')
+let gasPrice = 1000000000 // 1GWEI
+
+let _ = '        '
+
+contract('Token', async function(accounts) {
+  let token, metadata
+
+  before(done => {
+    ;(async () => {
+      try {
+        var totalGas = new BigNumber(0)
+
+        // Deploy Metadata.sol
+        metadata = await Metadata.new()
+        var tx = await web3.eth.getTransactionReceipt(metadata.transactionHash)
+        totalGas = totalGas.plus(tx.gasUsed)
+        console.log(_ + tx.gasUsed + ' - Deploy Metadata')
+        metadata = await Metadata.deployed()
+
+        // Deploy Token.sol
+        token = await Token.new("Token", "TKN", metadata.address)
+        var tx = await web3.eth.getTransactionReceipt(token.transactionHash)
+        totalGas = totalGas.plus(tx.gasUsed)
+        console.log(_ + tx.gasUsed + ' - Deploy Token')
+        token = await Token.deployed()
+
+        console.log(_ + '-----------------------')
+        console.log(_ + totalGas.toFormat(0) + ' - Total Gas')
+        done()
+      } catch (error) {
+        console.error(error)
+        done(false)
+      }
+    })()
+  })
+
+  describe('Token.sol', function() {
+    it('should pass', async function() {
+      assert(
+        true === true,
+        'this is true'
+      )
+    })
+  })
+})
+```
+
+Replace the dummy test called `should pass` inside the `describe` block to confirm the `tokenURI` returns strings of numbers correctly:
+
+```javascript
+it('should return metadata uints as strings', async function() {
+    const URI = 'https://domain.com/metadata/'
+
+    let tokenURI_uint = 12
+    let tokenURI_result = await token.tokenURI(tokenURI_uint)
+    assert(
+        URI + tokenURI_uint.toString() === tokenURI_result,
+        'incorrect value "' + tokenURI_result + '" returned'
+    )
+})
+```
+
+<!-- TODO: Fails, likely due to above -->
```

Comment:
@okwme Is this failing for me because I miss-understood what the 

> Also don't forget to set the deploy parameters for the token, including the metadata address.

step above meant?

Reply:
can i see your code somewhere? i have working version at https://github.com/okwme/block-workshop